easybuild/tools/modules.py

```diff
@@ -543,9 +546,25 @@ def mod_exists_via_show(mod_name):
 
             :param mod_name: module name
             """
-            mod_exists_regex = mod_exists_regex_template % re.escape(mod_name)
-            txt = self.show(mod_name)
-            return bool(re.search(mod_exists_regex, txt, re.M))
+            stderr = self.show(mod_name)
+            res = False
+            # Parse the output:
+            # - Skip whitespace
+            # - Any error -> Module does not exist
+            # - Check first non-whitespace line for something that looks like an absolute path terminated by a colon
+            mod_exists_regex = r'\s*/.+:\s*'
+            for line in stderr.split('\n'):
+                if OUTPUT_MATCHES['whitespace'].search(line):
+                    continue
+                if OUTPUT_MATCHES['error'].search(line):
+                    break
+                if re.match(mod_exists_regex, line):
+                    res = True
+                break
+            return res
+
+        if mod_exists_regex_template is not None:
+            self.log.deprecated('mod_exists_regex_template is no longer used', '5.0')
```

Comment:
Move this to the top of the method (above `stderr = self.show(mod_name)` please.

Reply:
That would be the inner method which is wrong. It should trigger from the outer method

Another point for collapsing this monstrosity and just use exists-via-show ;)

Reply:
Oh, ok, well, then move it above `def mod_exists_via_show` please